fix-point/src/CopyFiles.hs

```diff
@@ -0,0 +1,48 @@
+module CopyFiles where
+
+import Control.Applicative
+import Control.Exception (throw)
+import Control.Monad (forM_, unless, when)
+import Data.Text (pack)
+import System.Directory (copyFile, createDirectory, doesDirectoryExist,
+                         doesFileExist, listDirectory)
+import System.FilePath ((</>), takeExtension)
+
+import qualified Rename as R
+
+copyAll :: FilePath -> FilePath -> String -> IO ()
+copyAll source target newName = do
+    unlessM (doesDirectoryExist source) $
+        throw (userError "source does not exist")
+    whenM (doesFileOrDirectoryExist target) $
+        throw (userError "destination already exists")
+
+  -- if bottom two lines swapped, infinite creation of target occurs
+    content <- listDirectory source
+    let refinedContent = filter (/= "fix-point") content
+    createDirectory target
+    forM_ refinedContent $ \name -> do
```

Comment:
I think you can use `for_` function instead of `forM_` because it's one character shorter and looks slightly nicer:

* https://hackage.haskell.org/package/base-4.11.1.0/docs/Data-Foldable.html#v:for_